crates/typst/src/visualize/color.rs

```diff
@@ -1010,16 +1010,29 @@ impl Color {
         })
     }
 
-    /// Produces the negative of the color.
+    /// Produces the complementary color using a provided color space.
+    /// You can think of it as the opposite side on a color wheel.
+    ///
+    /// ```example
+    /// #square(fill: yellow)
+    /// #square(fill: yellow.negate())
+    /// #square(fill: yellow.negate(space: rgb))
+    /// ```
     #[func]
-    pub fn negate(self) -> Color {
-        match self {
+    pub fn negate(
+        self,
+        /// The color space used for the transformation. By default, a perceptual color space is used.
+        #[named]
+        #[default(ColorSpace::Oklab)]
+        space: ColorSpace,
```

Comment:
should this really default to oklab or rather to the color's space?

Reply:
My thinking was that eventually, all color transformation methods should take a `space` parameter defaulting to a perceptually accurate color space. This is already done for `mix` and `rotate`.
It would make `red.negate()` correspond very accurately with the expected complementary color for instance.
And, it means that `red.mix(red.negate())` produces gray, which is exactly what should happen.